dace/codegen/targets/cpp.py

```diff
@@ -1118,7 +1118,8 @@ def visit_Name(self, node: ast.Name):
         if name not in self.memlets:
             return self.generic_visit(node)
         memlet, nc, wcr, dtype = self.memlets[name]
-        if isinstance(dtype, dtypes.pointer):
+        if (isinstance(dtype, dtypes.pointer)
+                and memlet.subset.num_elements() == 1):
             return ast.Name(id="(*{})".format(name), ctx=node.ctx)
```

Comment:
How will this interact with GPU tasklets that are not devicelevel (e.g. ORT or cuBLAS)?
Dereferencing the pointer of a GPU_Global input would produce an error, right?

Reply:
This is meant for Python tasklets, where the behavior is actually to dereference (as Python has no notion of pointers)